core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import alluxio.PropertyKey;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Task for validating security configurations.
+ */
+public class SecureHdfsValidationTask implements ValidationTask {
+  private static final String LINE_SEPARATOR = System.getProperty("line.separator").toString();
+  private static final String ALLUXIO_BIN = "./bin/alluxio";
+  private static final String GETCONF_CMD = "getConf";
+  private static final Pattern PRINCIPAL_PATTERN =
+      Pattern.compile("(?<primary>[\\w][\\w-]*\\$?)(/(?<instance>[\\w]+))?(@(?<realm>[\\w]+))?");
+
+  private final String mProcess;
+
+  /**
+   * Constructor of {@link SecureHdfsValidationTask}.
+   *
+   * @param process type of the process on behalf of which this validation task is run
+   */
+  public SecureHdfsValidationTask(String process) {
+    mProcess = process.toLowerCase();
+  }
+
+  @Override
+  public boolean validate() {
+    // Check whether can login with specified principal and keytab
+    String principal = "";
+    String keytab = "";
+    String primary;
+    String instance;
+    String realm;
+    if (mProcess.equals("master")) {
+      principal = getResultFromProcess(new String[] {ALLUXIO_BIN, GETCONF_CMD,
+          PropertyKey.MASTER_PRINCIPAL.getName()}).getOutput();
+      keytab = getResultFromProcess(new String[] {ALLUXIO_BIN, GETCONF_CMD,
+          PropertyKey.MASTER_KEYTAB_KEY_FILE.getName()}).getOutput();
+    } else if (mProcess.equals("worker")) {
+      principal = getResultFromProcess(new String[] {ALLUXIO_BIN, GETCONF_CMD,
+          PropertyKey.WORKER_PRINCIPAL.getName()}).getOutput();
+      keytab = getResultFromProcess(new String[] {ALLUXIO_BIN, GETCONF_CMD,
+          PropertyKey.WORKER_KEYTAB_FILE.getName()}).getOutput();
+    }
+    Matcher matchPrincipal = PRINCIPAL_PATTERN.matcher(principal);
+    if (!matchPrincipal.matches()) {
+      System.err.format("Principal %s is not in the right format.%n", principal);
+      return false;
+    }
+    primary = matchPrincipal.group("primary");
+    instance = matchPrincipal.group("instance");
+    realm = matchPrincipal.group("realm");
+
+    // Login with principal and keytab
+    int exitVal =
+        getResultFromProcess(new String[] {"kinit", "-kt", keytab, principal}).getExitValue();
+    if (exitVal != 0) {
+      System.err.format("Kerberos login failed for %s with keytab %s with exit value %d.%n",
+          principal, keytab, exitVal);
+      System.err.format("Primary is %s, instance is %s and realm is %s.%n",
+          primary, instance, realm);
+      return false;
+    }
+    return true;
+  }
+
+  private ProcessExecutionResult getResultFromProcess(String[] args) {
```

Comment:
Might be useful to move this to `Utils.java`

Reply:
Done.